src/main/java/com/salesforce/dynamodbv2/mt/util/CachingAmazonDynamoDbStreams.java

```diff
@@ -175,12 +175,17 @@ public Builder withMaxIteratorCacheSize(int maxIteratorCacheSize) {
         }
 
         /**
-         * Disables the describe stream cache.
+         * Enables or disables the describe stream cache.
+         *
+         * @param describeStreamCacheEnabled Disable or enable the describe stream cache.
          * @return this Builder.
          */
-        public Builder withDescribeStreamCacheDisabled() {
-            LOG.info("describeStream cache is disabled");
-            this.describeStreamCacheEnabled = false;
+        public Builder withDescribeStreamCacheEnabled(boolean describeStreamCacheEnabled) {
+            this.describeStreamCacheEnabled = describeStreamCacheEnabled;
+
+            if (!this.describeStreamCacheEnabled) {
+                LOG.info("describeStream cache is disabled");
```

Comment:
IMHO, this doesn't belong in the builder, but rather in code calling it (since this class is more of a generic utility and some callers may not want to log this or log it differently).

Reply:
okay I'll take it out